Assignment6/CParkingLot.cpp

```diff
@@ -0,0 +1,95 @@
+#include "CParkingLot.h"
+
+
+CParkingLot::CParkingLot() {
+    std::cout << "How many places are in the parking lot?" << std::endl;
+    std::cin >> NumberOfPlaces_;
+
+    if (NumberOfPlaces_ > 0) {
+        std::cout << "Describe the parking lot" << std::endl;
+    }
+    get_parking_information(PlacesToCars_, CarsToPlaces_);
+
+    if (NumberOfPlaces_ > 0) {
+        std::cout << "How should we rearrange the parking lot?" << std::endl;
+    }
+    get_parking_information(ResultingPlacesToCars_, ResultingCarsToPlaces_);
+    ResultingZeroPlace_ = ResultingCarsToPlaces_[0];
+}
+
+
+void CParkingLot::get_parking_information(std::map<int, int>& PlacesToCars, std::map<int, int>& CarsToPlaces) {
+    int new_car;
+    for (int place = 0; place < NumberOfPlaces_; ++place) {
+        std::cin >> new_car;
+        PlacesToCars[place] = new_car;
+        CarsToPlaces[new_car] = place;
+    }
+}
+
+
+CParkingLot::CParkingLot(std::vector<int> &FirstState, std::vector<int> &SecondState) {
+    NumberOfPlaces_ = std::min(FirstState.size(), SecondState.size());
+    for(int place = 0; place < NumberOfPlaces_; ++place) {
+        PlacesToCars_[place] = FirstState[place];
+        CarsToPlaces_[FirstState[place]] = place;
+        ResultingPlacesToCars_[place] = SecondState[place];
+        ResultingCarsToPlaces_[SecondState[place]] = place;
+    }
+    ResultingZeroPlace_ = ResultingCarsToPlaces_[0];
+}
+
+
+void CParkingLot::rotate() {
```

Comment:
More blank lines :) it's hard to read a block of code with long names of variables and no blank lines :)